src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -0,0 +1,1447 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.Semantics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    partial class BoundExpression : IExpression
+    {
+        ITypeSymbol IExpression.ResultType => this.Type;
+
+        OperationKind IOperation.Kind => this.ExpressionKind;
+       
+        object IExpression.ConstantValue => this.ConstantValue?.Value;
+
+        SyntaxNode IOperation.Syntax => this.Syntax;
+        
+        protected virtual OperationKind ExpressionKind => OperationKind.None;
+        // protected abstract OperationKind ExpressionKind { get; }
```

Comment:
Eventually the commented-out declaration should be live and the one above deleted.